src/main/webapp/app/exercises/programming/manage/reset/programming-exercise-reset-dialog.component.ts

```diff
@@ -0,0 +1,128 @@
+import { ProfileService } from 'app/shared/layouts/profiles/profile.service';
+import { FeatureToggle } from 'app/shared/feature-toggle/feature-toggle.service';
+import { ProgrammingExercise } from 'app/entities/programming-exercise.model';
+import { ProgrammingExerciseResetOptions, ProgrammingExerciseService } from 'app/exercises/programming/manage/services/programming-exercise.service';
+import { faBan, faCircleNotch, faSpinner, faTimes } from '@fortawesome/free-solid-svg-icons';
+
+@Component({
+    selector: 'jhi-programming-exercise-reset-dialog',
+    templateUrl: './programming-exercise-reset-dialog.component.html',
+    styles: ['textarea { width: 100%; }'],
+})
+export class ProgrammingExerciseResetDialogComponent implements OnInit {
+    readonly FeatureToggle = FeatureToggle;
+
+    @Input() programmingExercise: ProgrammingExercise;
+
+    programmingExerciseResetOptions: ProgrammingExerciseResetOptions;
+    isLoading = false;
+    resetInProgress: boolean;
+    confirmText: string;
+
+    versionControlName?: string;
+    continuousIntegrationName?: string;
+
+    // Icons
+    faBan = faBan;
+    faCircleNotch = faCircleNotch;
+    faSpinner = faSpinner;
+    faTimes = faTimes;
+
+    constructor(
+        private alertService: AlertService,
+        private profileService: ProfileService,
+        private programmingExerciseService: ProgrammingExerciseService,
+        public activeModal: NgbActiveModal,
+    ) {}
+
+    /**
+     * Life cycle hook called by Angular to indicate that Angular is done creating the component
+     */
+    ngOnInit() {
+        this.isLoading = true;
+        this.profileService.getProfileInfo().subscribe((profileInfo) => {
+            if (profileInfo) {
+                this.versionControlName = profileInfo.versionControlName;
+                this.continuousIntegrationName = profileInfo.continuousIntegrationName;
+            }
+        });
+        this.resetInProgress = false;
+        this.programmingExerciseResetOptions = {
+            deleteBuildPlans: false,
+            deleteRepositories: false,
+            deleteParticipationsSubmissionsAndResults: false,
+            recreateBuildPlans: false,
+        };
+        this.isLoading = false;
+    }
+
+    /**
+     * Closes the active modal dialog and dismisses it with a 'cancel' reason
+     */
+    clear() {
+        this.activeModal.dismiss('cancel');
+    }
+
+    /**
+     * Determines whether to show the undeleted artifacts warning message
+     * @returns {boolean} true if the warning message should be displayed, false otherwise
+     */
+    showUndeletedArtifactsWarning() {
+        const options = [
+            this.programmingExerciseResetOptions.deleteBuildPlans,
+            this.programmingExerciseResetOptions.deleteRepositories,
+            this.programmingExerciseResetOptions.deleteParticipationsSubmissionsAndResults,
+        ];
+        return (!options[0] && (options[1] || options[2])) || (!options[1] && options[2]);
```

Comment:
I find this way of writing it a bit confusing. Maybe you could add some helper variable called `opts` and then write the whole conditions for clarity?
```suggestion
        const opts = this.programmingExerciseResetOptions;
        return (!opts.deleteBuildPlans && (opts. deleteRepositories || opts. deleteParticipationsSubmissionsAndResults)) || (!opts. deleteRepositories && opts. deleteParticipationsSubmissionsAndResults);
```

Reply:
I now have it in the html as a simplified method. The warning now only appears when option 3 is set but 2 is unset.